src/components/utils/src/file_system_qt.cc

```diff
@@ -0,0 +1,357 @@
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "utils/file_system.h"
+#include "utils/logger.h"
+#include "utils/string_utils.h"
+
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sstream>
+#include <QtCore>
+#include <QStorageInfo>
+#include <QDir>
+#include <QDebug>
+#include <QFileInfo>
+#include <QUrl>
+
+#include <io.h>
+// TODO(VS): lint error: Streams are highly discouraged.
+#include <fstream>
+#include <cstddef>
+#include <algorithm>
+
+#define R_OK    4
+#define W_OK    2
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "Utils")
+
+uint64_t file_system::GetAvailableDiskSpace(const std::string& path) {
+ QStorageInfo * mstor = new QStorageInfo(QString(path.c_str()));
+ qint64 b_aval = mstor->bytesAvailable();
+ if (b_aval != 0) {
+    return static_cast<uint64_t>(b_aval);
+  } else {
+    return 0;
+  }
+}
+
+int64_t file_system::FileSize(const std::string &path) {
+  if (file_system::FileExists(path)) {
+    return static_cast<uint64_t>(QFileInfo(QString(path.c_str())).size());
+  }
+  return 0;
+}
+
+size_t file_system::DirectorySize(const std::string& path) {
+    quint64 size = 0;
+     QFileInfo str_info(QString(path.c_str()));
+     if (str_info.isDir()){
+         QDir dir(QString(path.c_str()));
+         QFileInfoList list = dir.entryInfoList(QDir::Files | QDir::Dirs |  QDir::Hidden | QDir::NoSymLinks | QDir::NoDotAndDotDot);
+         for (int i = 0; i < list.size(); ++i) {
+             QFileInfo fileInfo = list.at(i);
+             if(fileInfo.isDir()){
+                 size += DirectorySize(fileInfo.absoluteFilePath().toStdString());
+             }
+             else {
+                 size += fileInfo.size();
+             }
+
+         }
+     }
+     return static_cast<size_t>(size);
+}
+
+std::string file_system::CreateDirectory(const std::string& name) {
+  if (!DirectoryExists(name)) {
+     QDir dir;
+     if(dir.mkdir(QString(name.c_str()))){
+         return name;
+     }
+  }
+  return "";
+}
+
+bool file_system::CreateDirectoryRecursively(const std::string& path) {
+   QDir dir;
+   return dir.mkpath(QString(path.c_str()));
+}
+
+bool file_system::IsDirectory(const std::string& name) {
+    QFileInfo checkFile(QString(name.c_str()));
+    return checkFile.isDir();
+}
+
+bool file_system::DirectoryExists(const std::string& name) {
+    QFileInfo checkFile(QString(name.c_str()));
+    if (checkFile.exists() && checkFile.isDir()) {
```

Comment:
Fix indent in all places. Here and in other methods. Please double check all methods.
Replace if block with one liner.